Platforms/QemuQ35Pkg/PlatformBuild.py

```diff
@@ -305,27 +322,67 @@ def FlashRomImage(self):
         startup_nsh = StartUpScriptManager()
         run_tests = (self.env.GetValue("RUN_TESTS", "FALSE").upper() == "TRUE")
         output_base = self.env.GetValue("BUILD_OUTPUT_BASE")
+        nshpath = os.path.join(output_base, "startup.nsh")
         shutdown_after_run = (self.env.GetValue("SHUTDOWN_AFTER_RUN", "FALSE").upper() == "TRUE")
         empty_drive = (self.env.GetValue("EMPTY_DRIVE", "FALSE").upper() == "TRUE")
+        drive_path = (self.env.GetValue("DRIVE_PATH", None))
+        drive_is_shared_dir = self.env.GetValue("USE_SHARED_DIR", None)
+        if drive_is_shared_dir is not None:
+            drive_is_shared_dir = (drive_is_shared_dir.upper() == "TRUE")
+        virtual_drive = None
+        ut = UnitTestSupport(os.path.join(output_base, "X64"))
+        test_regex = self.env.GetValue("TEST_REGEX", "")
+
+        if drive_path is not None: 
+            if not os.path.exists(drive_path):
+                raise Exception("DRIVE_PATH does not exist")
+            elif os.path.isdir(drive_path):
+                if drive_is_shared_dir != False:
+                    drive_is_shared_dir = True
+                elif drive_is_shared_dir is False:
+                    raise Exception("DRIVE_PATH is a directory, but USE_SHARED_DIR is set to FALSE")
+            elif os.path.basename(drive_path).endswith(".vhd"):
+                if drive_is_shared_dir != True:
+                    drive_is_shared_dir = False
+                    virtual_drive = VirtualDriveManager(drive_path, self.env)
+                else:
+                    raise Exception("DRIVE_PATH is a VHD file, but USE_SHARED_DIR is set to TRUE")
+            else:
+                raise Exception("DRIVE_PATH is not a directory or a VHD file")
+        elif drive_is_shared_dir is None:
+            drive_is_shared_dir = False

```

Comment:
This is messy, I'm suggesting a cleanup. I believe it hits what you're trying to do, but you'll need to verify, if you are willing to take on the suggestion.

```suggestion
        ut = UnitTestSupport(os.path.join(output_base, "X64"))
        test_regex = self.env.GetValue("TEST_REGEX", "")
        
        drive_path = (self.env.GetValue("DRIVE_PATH", None))
        drive_is_shared_dir = (self.env.GetValue("USE_SHARED_DIR", "FALSE") == "TRUE")
        virtual_drive = None

        if drive_path is not None: 
            if not os.path.exists(drive_path):
                raise Exception("DRIVE_PATH does not exist")
            elif os.path.isdir(drive_path) and drive_is_shared_dir is False:
                raise Exception("DRIVE_PATH is a directory, but USE_SHARED_DIR is set to FALSE")
            elif os.path.basename(drive_path).endswith(".vhd"):
                if drive_is_shared_dir is True:
                    raise Exception("DRIVE_PATH is a VHD file, but USE_SHARED_DIR is set to TRUE")
                virtual_drive = VirtualDriveManager(drive_path, self.env)                    
            else:
                raise Exception("DRIVE_PATH is not a directory or a VHD file")
```

Reply:
You could also be much more simple, but slightly more unsafe by doing what you did in QemuRunner.py

Reply:
Shareddir doesn't work well in the pipeline with many reads/writes so I'm removing the last several commits which added shareddir for testing